### PR TITLE
[Dto]: added dto for models

### PR DESCRIPTION
**Pull request by TarasYurkovskyi**
### Added dto for models:

**Accommodation entity:** 

- AccommodationResponseDto
- CreateAccommodationRequestDto
- UpdateAccommodationRequestDto

**Booking entity:** 

- BookingResponseDto
- CreateBookingRequestDto
- UpdateBookingRequestDto

**User entity**

- UserLoginRequestDto
- UserLoginResponseDto
- UserRegistrationRequestDto
- UserResponseDto
- Implemented FieldMatch annotation

### CODE REVIEW

**Review by MinoDentori — APPROVED** [review]
Good as for me

File: src/main/java/com/example/accommodationbookingapp/dto/booking/BookingResponseDto.java [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+import java.time.LocalDate;
+import lombok.Data;
+
+@Data
+public class BookingResponseDto {
+    private Long id;
+    private LocalDate checkInDate;
+    private LocalDate checkOutDate;
+    private Accommodation accommodationId;
+    private User userId;
```
Comment (annton-shakur) [inline src/main/java/com/example/accommodationbookingapp/dto/booking/BookingResponseDto.java:15]: Should be Long if you want to return ID, or accomodationresponsedto if you want to have dto inside dto. but i guess id would be simpler

File: src/main/java/com/example/accommodationbookingapp/dto/accommodation/UpdateAccommodationRequestDto.java [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+import jakarta.validation.constraints.Positive;
+import java.math.BigDecimal;
+import java.util.List;
+import lombok.Data;
+
+@Data
+public class UpdateAccommodationRequestDto {
+    private Accommodation.Type type;
+    private String location;
+    private String size;
```
Comment (annton-shakur) [inline src/main/java/com/example/accommodationbookingapp/dto/accommodation/UpdateAccommodationRequestDto.java:13]: i believe input data validation is required for update request as well, so it's better not to skip these fields

File: src/main/java/com/example/accommodationbookingapp/dto/booking/UpdateBookingRequestDto.java [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+
+import com.example.accommodationbookingapp.model.Accommodation;
+import java.time.LocalDate;
+import lombok.Data;
+
+@Data
+public class UpdateBookingRequestDto {
+    private LocalDate checkInDate;
+    private LocalDate checkOutDate;
+    private Accommodation accommodationId;
```
Comment (annton-shakur) [inline src/main/java/com/example/accommodationbookingapp/dto/booking/UpdateBookingRequestDto.java:11]: it should be Long here

**Review by annton-shakur — CHANGES_REQUESTED** [review]

File: src/main/java/com/example/accommodationbookingapp/dto/user/UserRegistrationRequestDto.java [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    @NotBlank
+    @Length(min = 6, max = 40)
+    private String password;
+
+    @NotBlank
+    @Length(min = 6, max = 40)
+    private String repeatPassword;
+
+    @NotBlank
+    @NotNull
```
Comment (annton-shakur) [inline src/main/java/com/example/accommodationbookingapp/dto/user/UserRegistrationRequestDto.java:26]: @NotNull is not required when you already have @NotBlank

File: src/main/java/com/example/accommodationbookingapp/dto/user/UserRegistrationRequestDto.java [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+    private String firstName;
+
+    @NotBlank
+    @NotNull
```
Comment (annton-shakur) [inline src/main/java/com/example/accommodationbookingapp/dto/user/UserRegistrationRequestDto.java:30]: @NotNull is not required when you already have @NotBlank

File: src/main/java/com/example/accommodationbookingapp/dto/booking/CreateBookingRequestDto.java [hunk 6]
```diff
@@ -0,0 +1,20 @@
+package com.example.accommodationbookingapp.dto.booking;
+
+import com.example.accommodationbookingapp.model.Accommodation;
+import com.example.accommodationbookingapp.model.Booking;
+import com.example.accommodationbookingapp.model.User;
```
Comment (vadymhrnk) [inline src/main/java/com/example/accommodationbookingapp/dto/booking/CreateBookingRequestDto.java:5]: Unused imports, check other files also

**Review by annton-shakur — CHANGES_REQUESTED** [review]
all good, just remove the extra annotations

**Review by vadymhrnk — CHANGES_REQUESTED** [review]

**Review by vadymhrnk — APPROVED** [review]
LGTM

**Review by annton-shakur — APPROVED** [review]
Well done!

**Pull request merged**